### PR TITLE
Cu 86bwmd18t/transaction history endpoint

### PR DESCRIPTION
**Pull request by polymathmode**
**Description of PR**
This feature queries the Transactions table & fetches the the five most recent transactions of a user.The response contains the details of each transaction including the users name, the amount transacted, the user's group title and the date of the transaction  ---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [x] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ ] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [ ] Does the endpoint return the correct success/error format?

---

**Screenshots**  
_add relevant screenshots here_

---

**Test Coverage**  
_add relevant screenshots here_

### CODE REVIEW

File: apps/server/src/models/transactions.ts [hunk 1]
```diff
@@ -90,8 +97,20 @@ Transactions.init(
...
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
```
Comment (KristofaJosh) [inline apps/server/src/models/transactions.ts:109]: I thought timestamps adds this already... why add it here again?

**Review by KristofaJosh — CHANGES_REQUESTED** [review]

File: apps/server/src/models/transactions.ts (hunk 1 above)
Comment (polymathmode) [reply apps/server/src/models/transactions.ts:109]: I could not access it  with  merely setting timestamps to 'true' though

File: apps/server/src/controllers/userControllers/getUserTransactionHistory.ts [hunk 2]
```diff
@@ -2,15 +2,34 @@ import { Response } from 'express'
...
+      walletId: transaction.walletId,
+      ownerId: transaction.ownerId,
+      ownerName: transaction.name,
+      action: transaction.action,
+      type: transaction.type,
+      receiverId: transaction.receiverId,
+      senderId: transaction.senderId,
+      date: transaction.createdAt,
+      amount: transaction.amount,
+    }))
```
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/getUserTransactionHistory.ts:31]: let's get the most recent transactions within the week

so

Sun - Sat

Comment (KristofaJosh) [reply apps/server/src/controllers/userControllers/getUserTransactionHistory.ts:31]: then maybe on request from the user, we send everything like asking for your bank statement.

Comment (KristofaJosh) [reply apps/server/src/controllers/userControllers/getUserTransactionHistory.ts:31]: I think we need more info:

Screenshot 2024-02-13 at 16 37 06

according to the UI, clicking on the card will most likely bring a popup

we can talk more here

File: apps/server/src/models/transactions.ts (hunk 1 above)
Comment (KristofaJosh) [reply apps/server/src/models/transactions.ts:109]: That's strange

**Review by KristofaJosh — CHANGES_REQUESTED** [review]

File: apps/server/src/controllers/userControllers/getUserTransactionHistory.ts (hunk 2 above)
Comment (polymathmode) [reply apps/server/src/controllers/userControllers/getUserTransactionHistory.ts]: ok

File: apps/server/src/controllers/userControllers/getUserTransactionHistory.ts [hunk 3]
```diff
@@ -1,16 +1,47 @@
...
+        // Query the database to get data within the current week
+        createdAt: { [Op.between]: [startDate, endDate] },
+      },
       order: [['createdAt', 'DESC']],
-      limit: 5,
+      include: [{ model: Wallets }, { model: Users }],
     })
+
+    // Extract relevant information for the response
+    const transactionHistory = userTransactions.map((transaction) => ({
```
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/getUserTransactionHistory.ts:32]: Let's paginate this at once

Ask @Don-christo  for help here:

or reference https://github.com/Ajo-Savings-Project/ajo-savings/blob/7243c8977e8979485ca7557397197cb7311d213e/apps/server/src/controllers/userControllers/getUpcomingActivities.ts#L103

**Review by KristofaJosh — CHANGES_REQUESTED** [review]

**Pull request merged**